src/asm_parse.cpp

```diff
@@ -133,7 +136,8 @@ static Deref deref(const std::string& width, const std::string& basereg, const s
     };
 }
 
-Instruction parse_instruction(const std::string& line, const std::map<std::string, label_t>& label_name_to_label) {
+InstructionOrConstraintsSet parse_instruction(const std::string& line,
+                                              const std::map<std::string, label_t>& label_name_to_label) {
```

Comment:
I think it's better to add an instruction (assertion) of the kind you want. Then do the check from inside ebpf_checker like any other check.

Reply:
And I think it's better to do it after #787

Reply:
I'm not sure I understand the scenario. Does it happen online or offline? Do you want to query the final result of the computation during the execution of the program?

Reply:
Yes. The BPF VM is single stepped through the instructions. After each instruction, a series of assertions is generated from the concrete state of the VM and checked against the abstract state at that label.

Divergence between the two signifies that either the VM is wrong or the verifier is wrong.

The majority of the verifier bugs I have been filing have been found using this technique (using BPF programs generated by specdif or libfuzzer).